daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -441,46 +441,36 @@ object SExpr {
   final case class ObserversDefRef(ref: DefinitionRef) extends SDefinitionRef
   final case class ToCachedContractDefRef(ref: DefinitionRef) extends SDefinitionRef
 
-  /** ImplementsDefRef(templateId, ifaceId) points to the Unit value if
-    * the template implements the interface.
-    */
-  final case class ImplementsDefRef(
-      templateId: TypeConName,
-      ifaceId: TypeConName,
-  ) extends SDefinitionRef {
-    override def ref = templateId;
-  }
-
-  /** CoImplementsDefRef(templateId, ifaceId) points to the Unit value if
-    * the interface provides an implementation for (co-implements) the template.
-    */
-  final case class CoImplementsDefRef(
-      templateId: TypeConName,
-      ifaceId: TypeConName,
-  ) extends SDefinitionRef {
-    override def ref = ifaceId;
-  }
-
-  /** ImplementsMethodDefRef(templateId, ifaceId, method) invokes the template's
-    * implementation of an interface method.
+  /** InterfaceInstanceDefRef(parent, interfaceId, templateId)
+    * points to the Unit value if 'parent' defines an interface instance
+    * of the interface for the template.
+    *
+    * invariants:
+    *   * parent == interfaceId || parent == templateId
+    *   * at most one of the following is defined:
+    *       InterfaceInstanceDefRef(i, i, t)
+    *       InterfaceInstanceDefRef(t, i, t)
+    *
+    * The parent is used to determine what package and module define
```

Comment:
@cocreature @remyhaemmerle-da, this is the reason we need the `parent` (previously, the reason we needed to override `ref`)